cmake/BLTPrivateMacros.cmake

```diff
@@ -248,6 +244,46 @@ macro(blt_setup_cuda_source_properties)
 endmacro(blt_setup_cuda_source_properties)
 
 
+##------------------------------------------------------------------------------
+## blt_split_source_list_by_language( SOURCES <sources>
+##                                    C_LIST <list name>
+##                                    Fortran_LIST <list name>)
+##------------------------------------------------------------------------------
+macro(blt_split_source_list_by_language)
+
+    set(options)
+    set(singleValueArgs C_LIST Fortran_LIST)
+    set(multiValueArgs SOURCES)
+
+    # Parse the arguments
+    cmake_parse_arguments(arg "${options}" "${singleValueArgs}" 
+                            "${multiValueArgs}" ${ARGN} )
+
+    # Check arguments
+    if ( NOT DEFINED arg_SOURCES )
+        message( FATAL_ERROR "Must provide a SOURCES argument to the 'blt_split_source_list_by_language' macro" )
+    endif()
+
+    # Generate source lists based on language
+    foreach(_file ${arg_SOURCES})
+        get_filename_component(_ext ${_file} EXT)
+
+        if(${_ext} IN_LIST BLT_C_FILE_EXTS)
+            if (${arg_C_LIST})
+                list(APPEND ${arg_C_LIST} ${_file})
+            endif()
+        elseif(${_ext} IN_LIST BLT_Fortran_FILE_EXTS)
+            if (${arg_Fortran_LIST})
+                list(APPEND ${arg_Fortran_LIST} ${_file})
+            endif()
+        else()
+            message(FATAL_ERROR "blt_split_source_list_by_language given source file with unknown file extension. Add the missing extension to the corresponding list (BLT_C_FILE_EXTS or BLT_Fortran_FILE_EXTS).\n Unknown file: ${_file}")
```

Comment:
Would we ever have a file in the source list that wouldn't categorize into C or Fortran ?
I.e. should there be a third list variable for the extra files?
(Possibly with a warning/error when such files exist?)